features/include/pcl/features/impl/normal_3d.hpp

```diff
@@ -53,48 +20,36 @@ pcl::NormalEstimation<PointInT, PointOutT>::computeFeature (PointCloudOut &outpu
   std::vector<float> nn_dists (k_);
 
   output.is_dense = true;
-  // Save a few cycles by not checking every point for NaN/Inf values if the cloud is set to dense
-  if (input_->is_dense)
-  {
-    // Iterating over the entire index vector
-    for (std::size_t idx = 0; idx < indices_->size (); ++idx)
-    {
-      if (this->searchForNeighbors ((*indices_)[idx], search_parameter_, nn_indices, nn_dists) == 0 ||
-          !computePointNormal (*surface_, nn_indices, output[idx].normal[0], output[idx].normal[1], output[idx].normal[2], output[idx].curvature))
-      {
-        output[idx].normal[0] = output[idx].normal[1] = output[idx].normal[2] = output[idx].curvature = std::numeric_limits<float>::quiet_NaN ();
-
-        output.is_dense = false;
-        continue;
-      }
 
-      flipNormalTowardsViewpoint ((*input_)[(*indices_)[idx]], vpx_, vpy_, vpz_,
-                                  output[idx].normal[0], output[idx].normal[1], output[idx].normal[2]);
-
-    }
-  }
-  else
-  {
+#pragma omp parallel for \
+  default(none) \
+  shared(output) \
+  firstprivate(nn_indices, nn_dists) \
+  num_threads(threads_) \
+  if (threads_ != -1)
```

Comment:
If `setNumberOfThreads()` isn't called and no value is supplied to the constructor, the algorithm won't be run? Shouldn't it just default to one thread?

Can we maybe add { } for better readability?

Reply:
Moving the if to be closer to the top might be better